Header header               # message header
string recipient_st         # name of recipient
string sender_st            # name of sender
float32 weight_f            # weight [0,1]